Plutarch/Verify.hs

```diff
@@ -0,0 +1,277 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+{-# OPTIONS_GHC -Wredundant-constraints #-}
+
+module Plutarch.Verify (
+  PTryFrom (PExcess, ptryFrom),
+  pcheckType,
+  pcheckByteStr,
+  pcheckInt,
+  pcheckList,
+  pcheckMap,
```

Comment:
Do we need to export these helper functions? Seem like they'd be used only rarely.

Reply:
I think so, as well, but @t1lde wanted something like `PTryFromMaybe` which would need to use those because otherwise the script would just fail upon using `pAsSomething`, so maybe, if we need that class that returns a maybe instead, we may as well need those?

Reply:
Do we really need `PTryFromMaybe`? I don't think we do, or at the very least, we shouldn't try to do it in this PR.

Reply:
Yeah, I'm pretty much of the same opinion, I don't really see the use case for not failing after a e.g. a redeemer has been dishonest.